build.py

```diff
@@ -0,0 +1,418 @@
+        while 1:
+            for cursor in "|/-\\":
+                yield cursor
+
+    def __init__(self, delay=None):
+        self.spinner_generator = self.spinning_cursor()
+        if delay and float(delay):
+            self.delay = delay
+
+    def spinner_task(self):
+        while self.busy:
+            sys.stdout.write(next(self.spinner_generator))
+            sys.stdout.flush()
+            time.sleep(self.delay)
+            sys.stdout.write("\b")
+            sys.stdout.flush()
+
+    def __enter__(self):
+        self.busy = True
+        threading.Thread(target=self.spinner_task).start()
+
+    def __exit__(self, exception, value, tb):
+        self.busy = False
+        time.sleep(self.delay)
+        if exception is not None:
+            return False
+
+
+def run_subprocess_with_spinner(name: str, proc: subprocess.Popen):
+    """
+    Opens a subprocess and redirect stdout and stderr to Python.\n
+    Shows a spinning Cursor while the command runs.\n
+    Exits with returncode of subprocess if not equals 0.\n
+    """
+    try:
+        p = None
+        # Register handler to pass keyboard interrupt to the subprocess
+
+        def handler(sig, frame):
+            print(f"================= {name} ABORTED BY USER ====================")
+            if p:
+                p.send_signal(signal.SIGINT)
+            else:
+                raise KeyboardInterrupt.add_note()
+
+        signal.signal(signal.SIGINT, handler)
+        with Spinner():
+            print(f"{name.rstrip('Ee')}ing...")
+            with proc as p:
+                for line in iter(p.stdout.readline, b""):
+                    # Print the line to the console
+                    sys.stdout.buffer.write(line)
+                for line in iter(p.stderr.readline, b""):
+                    # Print the error output to the console
+                    sys.stderr.buffer.write(line)
+                returncode = p.wait()
+                if returncode:
+                    print(
+                        f"===================== {name} FAILED ========================="
+                    )
+                    sys.exit(returncode)
+                else:
+                    print(
+                        f"===================== {name} SUCCESSFUL ====================="
+                    )
+    finally:
+        signal.signal(signal.SIGINT, signal.SIG_DFL)
+
+
+class Build:
+    def __init__(
+        self,
+        openstack: bool,
+        template: str,
+        conda_env: pathlib.Path,
+        provisioning: [str],
+        comment: str,
+        pvt_key: pathlib.Path,
+        ansible_args: str,
+    ):
+        self.openstack = openstack
+        self.template = template
+        self.comment = comment
+        self.pvt_key = pvt_key
+        self.provisioning = provisioning
+        self.ansible_args = ansible_args
+        self.image_name = self.assemble_name()
+        self.image_path = pathlib.Path(DIR_PATH + "/" + self.image_name + ".raw")
```

Comment:
```suggestion
        self.image_path = DIR_PATH / f"{self.image_name}.raw"
```

(only works if you converted DIR_PATH to a `pathlib.Path` object at the beginning)